### PR TITLE
:sparkles: feat: add hyren

### PR DESCRIPTION
**Pull request by FurryR**
# Proposed Changes

- Added [Hyren](https://github.com/FurryR/hyren) as an addons.

# Issues

- :warning: **docs** Documentation change & netlify change requested.
- :warning: **max size exceeded** `hyren` should be loaded asynchronously, while there are no APIs to do so.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 !/scripts
 !/test
 /addons/eureka/eureka
+/addons/hyren/hyren
```
Comment (SimonShiki) [inline .eslintignore]: add new line

File: package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
   "scripts": {
     "prepare": "husky install",
     "eureka:init": "cd ./addons/eureka/eureka && pnpm i && pnpm run build",
+    "hyren:init": "cd ./addons/hyren/hyren && git submodule update --init && cd scratch-render && npm ci && cd .. && npm ci && npm run build:injector",
```
Comment (SimonShiki) [inline package.json]: use pnpm instead

File: package.json [hunk 3]
```diff
@@ -67,4 +68,4 @@
       "eslint --fix"
     ]
   }
-}
+}
```
Comment (SimonShiki) [inline package.json]: add new line

**Review by SimonShiki — CHANGES_REQUESTED** [review]
for netlify build issues, I will resolve it until it can be merged.
Charlotte packs **all** addons into **one** script file so there's no way loading an addon asynchronously. The current plan is to separate addons from Charlotte itself and execute them by downloading them into IndexedDB (although I am not sure how safe this approach is)
If you really need this feature, you can wait until I finish this work before merging it.

File: package.json (hunk 2 above)
Comment (FurryR) [reply package.json]: as hyren is using `npm` as package manager (its scripts also used npm), im not sure about the side-effect of changing manager to `pnpm`.
I will still consider that.

**Review by SimonShiki — APPROVED** [review]

**Pull request merged**